src/model.js

```diff
@@ -1,62 +1,99 @@
-const isFunction = o => typeof o === 'function'
+const { assign, keys } = Object
 
-export const Node = funcOrType => body => ({ type: isFunction(funcOrType) ? funcOrType.name : funcOrType, ...body })
+//===============================================================================================================================
+// BEHAVIOUR
+//===============================================================================================================================
+
+const nodeBehaviour = {
+  is(typeOrCategory) {
+    return typeOrCategory.toString().split(',').some(type => this.type === type)
+  },
+
+  copy(diff) {
+    const clone = { ...this }
+    for (const key in diff) {
+      clone[key] = diff[key] instanceof Function ? diff[key](clone[key]) : diff[key]
+    }
+    return clone
+  }
+}
+
+export const traverse = configuration => assign(model => {
+  const applicableTxs = keys(configuration).filter(key => key.split(',').some(type => model.type === type))
+  if (applicableTxs.length < 1) throw new TypeError(`No matching traverse strategy for ${model.type} node`)
+  return configuration[applicableTxs[0]](model)
+}, configuration)
```

Comment:
Why do you copy the configuration to the lambda  function ?
I didn't get that part.
At least I haven't seen that used in the addDefaultConstructor function.
Maybe is to have backward compatibility with the compiler ?